src/gensxg.c

```diff
@@ -0,0 +1,444 @@
+#define _XOPEN_SOURCE
+#include <errno.h>
+#include <getopt.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <openssl/pem.h>
+
+#include "libsxg.h"
+
+static const struct option kOptions[] = {
+  { "help",         no_argument,       NULL, 'h' },
+  { "content",      required_argument, NULL, 'c' },
+  { "contentType",  required_argument, NULL, 't' },
+  { "header",       required_argument, NULL, 'H' },
+  { "miRecordSize", required_argument, NULL, 'm' },
+  { "url",          required_argument, NULL, 'u' },
+  { "certUrl",      required_argument, NULL, 'r' },
+  { "validityUrl",  required_argument, NULL, 'v' },
+  { "certificate",  required_argument, NULL, 'e' },
+  { "publicKey",    required_argument, NULL, 'p' },
+  { "privateKey",   required_argument, NULL, 'k' },
+  { "date",         required_argument, NULL, 'd' },
+  { "expire",       required_argument, NULL, 'x' },
+  { 0,              0,                 0,     0  },
+};
+
+static const char kHelpMessage[] =
+    "USAGE: gensxg [OPTIONS]\n\n"
+    "OPTIONS:\n"
+    "-help\n"
+    "  Show this message.\n"
+    "-content string\n"
+    "  Source to be used as payload of the SXG (default [./index.html]).\n"
+    "-contentType string\n"
+    "  Mime type of Source (default [text/html]).\n"
+    "-header string\n"
+    "  Custom inner header of the SXG. (optional)\n"
```

Comment:
* Mention this option can be specified multiple times?
* Also mention `Content-Type` should be specified via `-contentType`?

Reply:
done.